### PR TITLE
feat: search, page menus, pre-launch cleanup

### PR DESCRIPTION
**Pull request by misterbisson**
1. Search now works on front page and all other pages of the site, including in local development (fixes the bug seen in https://github.com/BluBracket/docs.blubracket.com/pull/53#issuecomment-1035192775)
2. Pages at the top of sections now list the descendant pages of those sections
3. Some general cleanup to make this more launch-ready

### CODE REVIEW

File: assets/js/index.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ Source:
 
   // https://discourse.gohugo.io/t/range-length-or-last-element/3803/2
 
-  {{ $list := (where .Site.Pages "Section" "docs") -}}
+  {{ $list := .Site.Pages -}}
```
Comment (misterbisson) [inline assets/js/index.js:97]: This fixes the bug seen in https://github.com/BluBracket/docs.blubracket.com/pull/53#issuecomment-1035192775

**Review by robbyandrews — APPROVED** [review]
Let's GTM

File: layouts/_default/list.html [hunk 2]
```diff
@@ -27,6 +27,7 @@
 			<h1>{{ .Title }}</h1>
 			<p class="lead">{{ .Params.lead | safeHTML }}</p>
 			{{ .Content }}
+			{{ partial "main/pages.html" . }}
```
Comment (misterbisson) [inline layouts/_default/list.html:30]: This adds a table of contents list to pages like https://docs.blubracket.com/how-to/ and https://docs.blubracket.com/faq/

File: layouts/partials/main/pages.html [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{{ with .Page.Pages }}
```
Comment (misterbisson) [inline layouts/partials/main/pages.html:1]: This is the new template partial that adds a table of contents list to pages like https://docs.blubracket.com/how-to/ and https://docs.blubracket.com/faq/

It's called from `layouts/_default/list.html`

**Pull request merged**